posthog/models/organization_domain.py

```diff
@@ -19,6 +21,40 @@ def verified_domains(self):
         # TODO: Verification becomes stale on Cloud if not reverified after a certain period.
         return self.exclude(verified_at__isnull=True)
 
+    def get_sso_enforcement_for_email_address(self, email: str) -> Optional[str]:
+        domain = email[email.index("@") + 1 :]
+        query = (
+            self.verified_domains()
+            .filter(domain=domain)
+            .exclude(sso_enforcement="")
+            .values("sso_enforcement", "organization__available_features")
+            .first()
+        )
+
+        if not query:
+            return None
+
+        candidate_sso_enforcement = query["sso_enforcement"]
+
+        # Check organization has a license to enforce SSO
+        if AvailableFeature.SSO_ENFORCEMENT not in query["organization__available_features"]:
+            print_warning(
+                [f"🤑🚪 SSO is enforced for domain {domain} but the organization does not have the proper license."]
+            )
+            return None
+
+        # Check SSO provider is properly configured
+        sso_providers = get_available_sso_providers()
+        if not sso_providers[candidate_sso_enforcement]:
+            print_warning(
+                [
+                    f"SSO is enforced for domain {domain} but the SSO provider ({candidate_sso_enforcement}) is not properly configured."
+                ]
+            )
+            return None
```

Comment:
I don't know if this is the right place to log this data.

The [backend coding conventions](https://posthog.com/handbook/engineering/conventions/backend-coding#logging) don't speak of `print_warning`, but more of `structlog`. CC @guidoiaquinti